packages/core/src/tasks/check-module-boundaries.ts

```diff
@@ -119,6 +121,46 @@ function findProjectGivenRoot(
   }
 }
 
+function includesWithWildcards(input: string, pattern: string): boolean {
+  if (pattern.includes('*')) {
+    const searchParts = pattern.split('*');
+    let lastIndex = 0;
+    for (const part of searchParts) {
+      const index = input.indexOf(part, lastIndex);
+      if (index === -1) {
+        return false;
+      }
+      lastIndex = index + part.length;
+    }
+    return true;
+  } else {
+    return input.includes(pattern);
+  }
+}
+
+declare global {
+  interface String {
+    includesWithWildcards(pattern: string): boolean;
+  }
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  interface Array<T> {
+    includesWithWildcards(pattern: string): boolean;
+  }
+}
+String.prototype.includesWithWildcards = function (
+  this: string,
+  pattern: string,
+): boolean {
+  return includesWithWildcards(this, pattern);
+};
+
+Array.prototype.includesWithWildcards = function (
+  this: string[],
+  pattern: string,
+): boolean {
+  return this.some((x) => includesWithWildcards(x, pattern));
+};
```

Comment:
2 things:

- let's not add things onto prototypes here, that gets out of control really fast and it's no harder to use the method directly. Extension methods in C# can be great, in TS it's frowned upon.

-  We should use minimatch for checking the pattern instead of a custom implementation. If you search the nx codebase for it you'll see some examples.

Reply:
RE: minimatch for pattern matching
Looking at [@nx/eslint-plugin](https://github.com/nrwl/nx/blob/9be869ff7bf7c39bfa05c35bb5cd9f73ba34ed2d/packages/eslint-plugin/src/utils/runtime-lint-utils.ts#L91C1-L91C71), I can hoist the regexp handling (globs are mapped to regexp).  I would assume this is preferred getting us closer to parity and not having to pull in the minimatch dependency?

Reply:
Eh, lets just go with minimatch. Honestly, thats what should have probably be done in our eslint plugin too.

Reply:
Nx itself depends on minimatch, so its not really bringing in an extra dep since the workspace would already have it.